.github/workflows/release.yml

```diff
@@ -0,0 +1,33 @@
+on:
+  release:
+    types: [published]
```

Comment:
I have one question about how you see the release process moving forward, we have some files that need updated each time a release is cut, e.g. CHANGELOG, pom.xml, README. Is the expectation that those files would get committed first and then the release is created?

It's not exactly cumbersome at the moment but those changes could probably be automated quite easily as well. In other projects we have a "pre-release" step where you can create a release candidate and then using that git tag it will update the needed files using the tag e.g. `1.9.0rc1` would bump the pom.xml, and any description in the tag that gets created is used for release notes. Curious if that could be a followup for this PR or if you see the release process being a bit different with this PR.

Reply:
I agree. I was trying to find a place to stop and call it good enough for now, but I'd like to get to where the tags are enough. My thought is that this takes care of most of the struggle with releasing, but with the next release we can collapse develop and main, publish snapshot releases, and parameterize the version.

Reply:
OK great this is definitely the most cumbersome steps being automated which is huge - thanks David!